px-popover.html

```diff
@@ -270,7 +270,8 @@
                 if( this._isDescendant(this.$.popoverWrapper, event.target) ) {
                     return false;
                 } else {
-                    if(event.target.id === this.for) {
+                    if ((typeof (this.for) === 'string' && event.target.id === this.for) ||
```

Comment:
I see a few issues here that need to be addressed if we want to explicitly support the `for` property being a reference to an HTMLElement rather than an ID. While some of the code in px-popover supports that now, it was never a documented behavior, so we'll need to be careful if we're going to create a new API functionality that must be supported in future versions:

**1. I don't think we want to compare `this.for` to `event.target`, I think we want to compare it to `event.currentTarget`**

When we bind the popover to a target through `for`, we want to react to click events on that target or any of its children. Click events bubble through the DOM, so clicking on a child element triggers the click event listener on all of its parent elements until the event hits the window. Example:

```
<div id="parent">
  <div id="child">
    <button>Target Button</button>
  </div>
</div>
```

If we want to set the px-popover `for` target to `div#parent`, click events on the child button should also trigger the popover. 

If we only used the `_toggle` method to handle click events on the `for` target, we wouldn't need to check anything. A click event handled by `_toggle` would definitely be on the `for` target, so we could just do whatever necessary respond to it. But we use `_toggle` to handle click events on the `for` target AND on the px-popover element, and treat each kind of click differently.

I think we should be checking if `this.for === event.currentTarget` to determine if we want to handle the click as a click on the `for` target. You'll need to do some testing on a DOM structure like the example above to let me know if that's the case. This also needs to be tested in browsers without native shadow DOM (IE11) and browsers with native shadow DOM (Chrome). Please add automated tests to cover this functionality.

**2. We should check that `this.for` is an object and an HTMLElement**

On https://github.com/PredixDev/px-popover/pull/20/files#diff-2e094ab5aba394b9f899b4166b1a436eR273 and on https://github.com/midgetspy/px-popover/blob/f350841c98514f79a639c202df46bad95bc95f6a/px-popover.html#L187. Just because it's an object doesn't mean we can bind event listeners and find it in the DOM. We should add a check of `this.for instanceof HTMLElement` after the typeof check.

Reply:
These are absolutely things that should be discussed before official support could be added for HTMLElement. That is, however, not the purpose of this PR; this PR is only meant to fix a bug in the existing **undocumented** behavior.

Using `event.target` in `_toggle()` is how the existing code works and changing the existing behavior is not in the scope of this PR.

I will update my PR with a check to ensure that the object is an HTMLElement.